bull/pdl-queue/consumer.js

```diff
@@ -28,33 +28,36 @@ async function getZipAndBytelength(no_of_pages, id, title, job) {
   title = title.replace(/ /g, "_");
   var img = zip.folder(`${title}_images`);
   let temp_pages = no_of_pages;
+  let downloadImageStatus;
   var download_image = async function (uri, filename) {
-    await rp({
-      method: "GET",
-      uri,
-      encoding: null,
-      transform: function (body, response) {
-        return { headers: response.headers, data: body };
-      },
-    })
-      .then(async function (body) {
-        if (/image/.test(body.headers["content-type"])) {
-          var data = new Buffer(body.data);
-          img.file(filename, data.toString("base64"), { base64: true });
-        }
-      })
-      .catch(function (err) {
-        --no_of_pages;
+    try {
+      const body = await rp({
+        method: "GET",
+        uri,
+        encoding: null,
+        transform: function (body, response) {
+          return { headers: response.headers, data: body };
+        },
       });
+      if (/image/.test(body.headers["content-type"])) {
+        var data = new Buffer(body.data);
+        img.file(filename, data.toString("base64"), { base64: true });
+      }
+      return 200;
+    } catch (err) {
+      --no_of_pages;
+      return err.statusCode;
+    }
   };
   for (let i = 1; i <= temp_pages; ++i) {
     const str = `http://www.panjabdigilib.org/images?ID=${id}&page=${i}&pagetype=null&Searched=W3GX`;
-    await download_image(str, `${title}_${i}.jpeg`);
+    downloadImageStatus = await download_image(str, `${title}_${i}.jpeg`);
     job.progress(Math.round((i / temp_pages) * 82));
   }
+  let isError = downloadImageStatus === 500 ? true : false;
```

Comment:
What if the error is not 500 but 503 or 404 or something else? We should handle all the possible failures. I suggest check for 2xx, else return failure.

Reply:
@coderwassananmol  I have made the requested changes